worksheet.rb

```diff
@@ -31,4 +122,76 @@
 # - the total amount of money each driver has made
 # - the average rating for each driver
 # - Which driver made the most money?
-# - Which driver has the highest average rating?
\ No newline at end of file
+# - Which driver has the highest average rating?
+
+def num_of_rides(driver_rides)
+  return driver_rides.length
+end
+
+def total_earned(driver_rides)
+  total_earned = driver_rides.map.sum { |ride| ride[:cost]}
+
+  return total_earned
+end
+
+def average_rating(driver_rides)
+  total_rating = driver_rides.map.sum { |ride| ride[:rating]}
+
+  average = (total_rating.to_f / num_of_rides(driver_rides)).round(1)
+
+  return average
+end
+
+puts "\n*DRIVER SUMMARIES*\n"
+puts "~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+
+# stores each driver's data regarding total rides, total earned, and average rating
+driver_summaries = []
+
+# invoke methods, finding the total rides, total earned, and average rating per driver
+# stores and prints out each driver's data
+ride_share_drivers.each do |driver, rides|
+  hash_driver = {}
+
+  hash_driver[:driver_id] = driver
+  puts "\nDriver ID: #{driver}"
+
+  hash_driver[:total_rides] = num_of_rides(rides)
+  puts "Total Rides: #{hash_driver[:total_rides]}"
+
+  hash_driver[:total_earned] = total_earned(rides)
+  puts "Total Earned: $#{'%.2f' %hash_driver[:total_earned]}"
+
+  hash_driver[:average_rating] = average_rating(rides)
+  puts "Average Rating: #{hash_driver[:average_rating]}\n\n"
+  puts "~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+
+  # each driver's data will be stored into the array, driver_summaries
+  driver_summaries << hash_driver
+end
```

Comment:
Consider moving this code block to after your remaining method definitions so that all method definitions are grouped together, and all code that creates output is grouped together.